### PR TITLE
Sphinx docs

### PR DESCRIPTION
**Pull request by stas**
This is a WIP branch on the documentation review and migration to Sphinx.
I will take some more time to review what is currently in our wiki and add relevant pages to the `docs`.

A preview of this is currently available at:
http://aleph.readthedocs.io/en/latest/

I will let you know when a final-ish review is needed.

### CODE REVIEW

**Comment by stas** [comment]
Alright, this is ready for a final review.
I've moved all the pages I found on wiki.
Some of the articles went through minor edits.

I'd like to ask about one last things, which I think should be outside the docs scope and can be automated.
1. Move pending entity deletion into a `manage.py` task: [Maintenance#deleting-all-pending-entities](https://github.com/pudo/aleph/wiki/Maintenance#deleting-all-pending-entities)
2. Review the indexes need for performance and add those as part of a migration [Maintenance#adding-indexes-for-performance](https://github.com/pudo/aleph/wiki/Maintenance#adding-indexes-for-performance)

**Comment by stas** [comment]
Oh, and we should have now the automated API docs:
http://aleph.readthedocs.io/en/latest/autoapi/index.html

**Comment by pudo** [comment]
God I love this. Will try to read through it when I'm more awake tomorrow, but it looks like an impressive amount of work.

File: docs/usage.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+Below are some of common use cases Aleph was designed to cover. To learn more
+about the needs of such a tool, please refer to the notes about the
+[user needs in investigative journalism](http://www.influencemapping.org/workshop/user_needs.html).
+A lot more resources are also available on the
+[Technology for Investigative Journalism Conference](http://www.influencemapping.org/workshop/) webpage.
+
+There's also a [glossary](glossary.md) describing the keywords used in Aleph.
+
+Consider some common use-cases like:
```
Comment (pudo) [inline docs/usage.md]: I like that you added the use cases into the usage document, that really helps make it more plastic

**Pull request merged**